### PR TITLE
fix(recovery_codes): if user has no totp setup then redirect to totp setup instead of recovery_codes

### LINKED ISSUE #6186: Can navigate to recovery codes even if you do not have TOTP setup
Found this in https://github.com/mozilla/fxa-content-server/pull/6181. Users should not be able to directly navigate to `/settings/two_step_authentication/recovery_codes`, if they do not have TOTP setup. We should redirect them to TOTP setup page instead.

### PR DESCRIPTION
**Pull request by farhan787**
fixes #6186

### CODE REVIEW

**Comment by farhan787** [comment]
@vbudhram please review it :-)

File: app/scripts/views/settings/recovery_codes.js [hunk 1]
```diff
@@ -97,6 +97,19 @@ const View = FormView.extend({
   },
 
   setInitialContext(context) {
+    // if user has no totp setup then the user should not be able to
+    // directly navigate to /settings/two_step_authentication/recovery_codes
+    const that = this;
```
Comment (vbudhram) [inline app/scripts/views/settings/recovery_codes.js]: We normally use `self` variable name when referencing `this`, or use the `this` directly. I think in this case you can just use `this directly below and use use an intermediate value.

File: app/scripts/views/settings/recovery_codes.js [hunk 2]
```diff
@@ -97,6 +97,19 @@ const View = FormView.extend({
...
+    const account = this.getSignedInAccount();
+
+    account.checkTotpTokenExists()
```
Comment (vbudhram) [inline app/scripts/views/settings/recovery_codes.js]: I would pull this logic into a `beforeRender` function. This function gets called automatically to setup required things before the view is show. Example, https://github.com/mozilla/fxa-content-server/blob/58b8eaac1819f490be9c5b90b7d3d4b72742268e/app/scripts/views/reset_password.js#L45.

File: app/tests/spec/views/settings/recovery_codes.js [hunk 3]
```diff
@@ -82,6 +83,12 @@ describe('views/settings/recovery_codes', () => {
       verified: true
     });
 
+    sinon.stub(account, 'checkTotpTokenExists').callsFake(() => {
```
Comment (vbudhram) [inline app/tests/spec/views/settings/recovery_codes.js]: We will want to test two scenarios. 

1. `navigate` gets called with `settings/two_step_authentication` when checkTotpTokenExists is true
2. `navigate` does not get called when checkTotpTokenExists is false

**Comment by farhan787** [comment]
@vbudhram I hope you like the changes :-)

File: app/tests/spec/views/settings/recovery_codes.js [hunk 4]
```diff
@@ -91,10 +96,41 @@ describe('views/settings/recovery_codes', () => {
...
+        hasToken = true;
+
+        return view.beforeRender();
+      });
+
+      it('should not redirect to `settings/two_step_authentication`', () => {
+        assert.equal(view.navigate.callCount, 0);
+      });
+
+      it('should show recovery codes, translated success message', () => {
```
[the remark this answers is not in the document]
Comment (farhan787) [reply app/tests/spec/views/settings/recovery_codes.js]: sure @vbudhram

**Comment by farhan787** [comment]
@vbudhram thanks :-)

**Pull request merged**